### PR TITLE
newrelicexporter: multi-tenant support for sending trace data + perf enhancements

### PR DESCRIPTION
**Pull request by alanwest**
* Adds `APIKeyHeader` config option enabling reading New Relic API key from request header.
* `spans_url_override` and `metrics_url_override` has been renamed to `spans_host_override` and `metrics_host_override` to reflect the fact that you only need to specify `host[:port]` when overriding these endpoints.
* Updates reference to `newrelic-telemetry-sdk-go` which includes a number of performance improvements.
* Make communication with New Relic's Trace API directly rather than through the harvester provided through the `newrelic-telemetry-sdk-go`.

@a-feld @nrcventura @jack-berg

### CODE REVIEW

File: exporter/newrelicexporter/factory.go [hunk 1]
```diff
@@ -30,8 +31,7 @@ func NewFactory() component.ExporterFactory {
 	return exporterhelper.NewFactory(
 		typeStr,
 		createDefaultConfig,
-		exporterhelper.WithTraces(createTraceExporter),
-		exporterhelper.WithMetrics(createMetricsExporter))
```
Comment (nrcventura) [inline exporter/newrelicexporter/factory.go:34]: This line does have the effect of enabling the metrics exporter, so we'll likely want to conditionally enable the various exporters.

Comment (alanwest) [reply exporter/newrelicexporter/factory.go]: Added `NEW_RELIC_NO_METRIC_SUPPORT` as a means for us to disable metric support for now.

File: exporter/newrelicexporter/factory.go [hunk 2]
```diff
@@ -50,12 +61,14 @@ func createTraceExporter(
...
-	exp, err := newExporter(params.Logger, cfg)
+	exp, err := newTraceExporter(params.Logger, cfg)
 	if err != nil {
 		return nil, err
 	}
 
-	return exporterhelper.NewTraceExporter(cfg, params.Logger, exp.pushTraceData, exporterhelper.WithShutdown(exp.Shutdown))
+	// The logger is only used in a disabled queuedRetrySender, which noisily logs at
+	// the error level when it is disabled and errors occur.
+	return exporterhelper.NewTraceExporter(cfg, zap.NewNop(), exp.pushTraceData)
```
Comment (nrcventura) [inline exporter/newrelicexporter/factory.go:71]: A reminder for us to resolve this difference.

File: exporter/newrelicexporter/newrelic.go [hunk 3]
```diff
@@ -98,18 +165,50 @@ func (e exporter) pushTraceData(ctx context.Context, td pdata.Traces) (int, erro
...
-					continue
-				}
+				spans = append(spans, nrSpan)
 				goodSpans++
 			}
+			batch.Spans = append(batch.Spans, spans...)
 		}
 	}
+	batches := []telemetry.PayloadEntry{&batch}
+	insertKey := e.extractInsertKeyFromHeader(ctx)
```
Comment (nrcventura) [inline exporter/newrelicexporter/newrelic.go:175]: We probably should check to see if the config says we should extract the key from the header before attempting to do this.

Comment (alanwest) [reply exporter/newrelicexporter/newrelic.go:171]: `extractInsertKeyFromHeader` already does this check.

**Review by a-feld — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**